SX1276/SX1276_LoRaRadio.h

```diff
@@ -215,10 +215,26 @@ class SX1276_LoRaRadio: public LoRaRadio {
      *  setting the receiver in continuous mode is by using set_rx_config()
      *  API.
      *
+     *  NOTE: This API will be deprecated
+     *
      *  @param timeout       Reception timeout [ms]
      *
      */
-    virtual void receive(uint32_t timeout);
+    MBED_DEPRECATED_SINCE("mbed-os-5.10", "receive(uint32_t) will be deprecated. "
```

Comment:
You can't warn about deprecation in the future. It's either deprecated or it isn't. This is saying "I'm telling you that I'm going to tell you that you shouldn't use this in the future, but I'm not telling you not to use it now", which doesn't make a lot of sense.

Anyway, a deprecation notice isn't appropriate for this outside-mbed OS driver implementation. It has to provide the current method, which is `receive(timeout)`, but is providing `receive()` ready for some future version of `LoRaRadio`. (Which I still think could arrive by 5.10, once all drivers are updated).